tests/smoktest/base_smoketest.py

```diff
@@ -0,0 +1,215 @@
+        run(f"{cls.venv_bin}pip install --upgrade .", cwd="dcp-cli")
+
+        # Configure the CLI
+        #
+        cli_config = {"DSSClient": {"swagger_url": f"https://{os.environ['API_DOMAIN_NAME']}/v1/swagger.json"}}
+        cli_config_filename = f"{cls.workdir.name}/cli_config.json"
+        with open(cli_config_filename, "w") as fh2:
+            fh2.write(json.dumps(cli_config))
+        os.environ["HCA_CONFIG_FILE"] = f"{cls.workdir.name}/cli_config.json"
+
+    def upload_bundle(self, replica, bucket, src_dir):
+        """ uploads bundle to DSS, returns response from CLI """
+        create_res = run_for_json(f"{self.venv_bin}hca dss upload "
+                                  f"--replica {replica.name} "
+                                  f"--staging-bucket {bucket} "
+                                  f"--src-dir {src_dir}")
+        return create_res
+
+    def get_bundle(self, replica, bundle_uuid):
+        """ returns bundle manifest from DSS"""
+        return run_for_json(f"{self.venv_bin}hca dss get-bundle --replica {replica.name}"
+                            f" --uuid {bundle_uuid}")
+
+    def _test_get_bundle(self, replica, bundle_uuid):
+        """ tests that a bundle can be downloaded"""
+        download_res = self.get_bundle(replica, bundle_uuid)
+        self.assertEqual(bundle_uuid, download_res['bundle']['uuid'])
+
+    def checkout_initiate(self, replica, bundle_uuid):
+        """ initiates a bundle checkout, return the checkout job"""
+        res = run_for_json(f"{self.venv_bin}hca dss post-bundles-checkout --uuid {bundle_uuid} "
+                           f"--replica {replica.name}")
+        print(f"Checkout jobId: {res['checkout_job_id']}")
+        return res
+
+    def _test_checkout(self, starting_replica, bundle_uuid, bundle_version,
+                       checkout_job_id, checkout_bucket, file_count):
+        """ asserts that the checkout mechanism is operational """
+        for i in range(10):
+            res = run_for_json(f"{self.venv_bin}hca dss get-bundles-checkout --checkout-job-id {checkout_job_id} "
+                               f"--replica {starting_replica.name}")
+            status = res['status']
+            self.assertGreater(len(status), 0)
+            if status == 'RUNNING':
+                time.sleep(6)
+            else:
+                self.assertEqual(status, 'SUCCEEDED')
+                blob_handle = self.get_blobstore(starting_replica)
+                object_key = get_dst_bundle_prefix(bundle_uuid, bundle_version)
+                print(f"Checking bucket {checkout_bucket} "
+                      f"object key: {object_key}")
+                files = list(blob_handle.list(checkout_bucket, object_key))
+                self.assertEqual(len(files), file_count)
+                break
+        else:
+            self.fail("Timed out waiting for checkout job to succeed")
+
+    def post_search_es(self, replica, es_query):
```

Comment:
I know this is a deviation from the original smoketest, but I'm realizing that we test some `es_query` functionality but there's no reciprocal `jmespath_query` test.  Since we're promoting `jmespath_query` as the default and intended user experience, should we add those tests symmetrically?

Reply:
We should, but in a different PR